name.abuchen.portfolio/src/name/abuchen/portfolio/online/Column.java

```diff
@@ -0,0 +1,166 @@
+    {
+        @Override
+        protected DecimalFormat initialValue()
+        {
+            DecimalFormatSymbols unusualSymbols = new DecimalFormatSymbols(Locale.US);
+            unusualSymbols.setGroupingSeparator('\'');
+            return new DecimalFormat("#,##0.##", unusualSymbols); //$NON-NLS-1$
+        }
+    };
+
+    private final Pattern[] patterns;
+
+    protected Column(String[] strings)
+    {
+        this.patterns = new Pattern[strings.length];
+        for (int ii = 0; ii < strings.length; ii++)
+            this.patterns[ii] = Pattern.compile(strings[ii]);
+    }
+
+    public boolean matches(Element header)
+    {
+        String text = header.text();
+        for (Pattern pattern : patterns)
+        {
+            if (pattern.matcher(text).matches())
+                return true;
+        }
+        return false;
+    }
+
+    public abstract void setValue(Element value, LatestSecurityPrice price, String languageHint) throws ParseException;
+
+    protected long asQuote(Element value, String languageHint) throws ParseException
+    {
+        String text = value.text().trim();
+
+        DecimalFormat format = null;
+
+        if ("de".equals(languageHint)) //$NON-NLS-1$
+            format = DECIMAL_FORMAT_GERMAN.get();
+        else if ("en".equals(languageHint)) //$NON-NLS-1$
+            format = DECIMAL_FORMAT_ENGLISH.get();
+
+        if (format == null)
+        {
+            // check first for apostrophe
+
+            int apostrophe = text.indexOf('\'');
+            if (apostrophe >= 0)
+                format = DECIMAL_FORMAT_APOSTROPHE.get();
+        }
+
+        if (format == null)
+        {
+            // determine format based on the relative location of the last
+            // comma and dot, e.g. the last comma indicates a German number
+            // format
+            int lastDot = text.lastIndexOf('.');
+            int lastComma = text.lastIndexOf(',');
+            format = Math.max(lastDot, lastComma) == lastComma ? DECIMAL_FORMAT_GERMAN.get()
+                            : DECIMAL_FORMAT_ENGLISH.get();
+        }
+
+        double quote = format.parse(text).doubleValue();
+        return Math.round(quote * Values.Quote.factor());
+    }
+    
+    private Hashtable<String,Integer> symbols = new Hashtable<String,Integer>() {
+        {
+            put("k",       1000);
+            put("K",       1000);
+            put("T",       1000);
+            put("M",    1000000);
+        }
+    };
+
+    protected int asInt(Element elem) throws ParseException
+    {
+        String text = elem.text().trim();
+        int multiplier = 1;
+        double value = -0.1;
+        try
+        {
+            Pattern numberPattern = Pattern.compile("([0-9,.]+)");
+            Matcher numberMatcher = numberPattern.matcher(text);
+            if (numberMatcher.find())
+            {
+                String match = numberMatcher.group(1);
```

Comment:
This "lastDot" and "lastComma" unfortunately does not really work well.
I had similar logic for the quote initially. But what if the value is "1,045". It could be read as American format (one thousand and forty five) or as German format (one euro and 4 and half cents).

That's why "asQuote" also uses the "langaugeHint" which is basically the language given in the HTML page. I guess the volumne should also use that.